### PR TITLE
[ASCII-1158] Add unit tests for the api component

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?
As part of the component framework effort, we are going to refactor how we register routes to the internal api server of the agent similar to the status component.

The api server has a lot of dependencies so this PR adds some tests around the component and registers routes before I begin the refactoring.

### CODE REVIEW

File: comp/core/flare/component_mock.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+package flare
+
+// team: agent-shared-components
+
+// Mock is the mocked component type.
+type Mock interface {
+	Component
+
+	// no further methods are defined.
+}
```
Comment (jeremy-hanna) [inline comp/core/flare/component_mock.go:17]: Added some missing component mocks that the api server has as dependencies

File: comp/metadata/inventorychecks/inventorychecksimpl/mock.go [hunk 2]
```diff
@@ -51,8 +52,9 @@ func (m *InventorychecksMock) GetInstanceMetadata(instanceID string) map[string]
 //
 //	fxutil.Test[dependencies](
 //	   t,
-//	   inventorychecks.MockModule,
+//	   inventorychecks.MockModule(),
 //	)
-var MockModule = fxutil.Component(
-	fx.Provide(NewMock),
-)
+func MockModule() fxutil.Module {
```
Comment (jeremy-hanna) [inline comp/metadata/inventorychecks/inventorychecksimpl/mock.go:57]: This wasn't used so I updated the interface to be consistent with the other component mock functions

File: comp/forwarder/eventplatformreceiver/eventplatformreceiverimpl/mock.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	return true
+}
+
+func (epr *MockEventPlatformReceiver) HandleMessage(m *message.Message, rendered []byte, eventType string) {
+}
+
+func (epr *MockEventPlatformReceiver) Filter(filters *diagnostic.Filters, done <-chan struct{}) <-chan string {
+	c := make(chan string)
+	return c
+}
```
Comment (jeremy-hanna) [inline comp/forwarder/eventplatformreceiver/eventplatformreceiverimpl/mock.go:39]: I did not spend too much time with the actual mocked behavior here as I am not very familiar with this code, let me know if I should improve it - I really just needed the component initialization for fx

File: comp/api/api/apiimpl/api_test.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	))
+	tags := fxutil.Test[tagger.Mock](t, tagger.MockModule())
+	ac := autodiscoveryimpl.CreateMockAutoConfig(t, nil)
+	sender := aggregator.NewNoOpSenderManager()
+	log := fxutil.Test[optional.Option[logsAgent.Component]](t,
+		fx.Provide(func() optional.Option[logsAgent.Component] {
+			return optional.NewNoneOption[logsAgent.Component]()
+		}),
+	)
+	col := fxutil.Test[optional.Option[collector.Component]](t,
```
Comment (ogaca-dd) [inline comp/api/api/apiimpl/api_test.go:107]: It is recommended to have a single call to `fxutil.Test` per function to make sure the components are not instanciated several times. One easy way to do that is to have a struct `deps` that contains all the components you need to your test and create a `deps` using `fxutil.Test[des](...)`.

File: comp/api/api/apiimpl/internal/agent/agent_test.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+func TestSetupHandlers(t *testing.T) {
+	router := setupRoutes(t)
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	res, err := http.Get(ts.URL + "/version")
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
```
Comment (ogaca-dd) [inline comp/api/api/apiimpl/internal/agent/agent_test.go:157]: This is equivalent to the one line `require.NoError(t, err)`

**Review by ogaca-dd — APPROVED** [review]

File: comp/api/api/apiimpl/api_test.go (hunk 4 above)
Comment (GustavoCaso) [reply comp/api/api/apiimpl/api_test.go:107]: We could probably have a lint that would prevent developers from calling `fxutil.Test` multiple times on their tests

File: comp/api/api/apiimpl/api_test.go [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+			}),
+			fetchonlyimpl.MockModule(),
+		),
+	)
+}
+
+func getTestAPIServer(t *testing.T) *apiServer {
+	p := getProvides(t)
+	return p.(*apiServer)
+}
```
Comment (GustavoCaso) [inline comp/api/api/apiimpl/api_test.go:88]: Is this necessary?

File: comp/api/api/apiimpl/api_test.go [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+				return optional.NewNoneOption[rcserviceha.Component]()
+			}),
+			fetchonlyimpl.MockModule(),
+		),
+	)
+}
+
+func getTestAPIServer(t *testing.T) *apiServer {
+	p := getProvides(t)
+	return p.(*apiServer)
```
Comment (GustavoCaso) [inline comp/api/api/apiimpl/api_test.go:87]: Do we need to return the private struct rather the component? Are we not testing the component functions?

**Review by GustavoCaso — APPROVED** [review]
Thanks for writing tests 🎉

File: comp/api/api/apiimpl/api_test.go (hunk 7 above)
Comment (jeremy-hanna) [reply comp/api/api/apiimpl/api_test.go:87]: this is [how it is implemented](https://github.com/DataDog/datadog-agent/blob/main/comp/api/api/apiimpl/api.go#L87) currently, I wanted to get tests around it before making changes

File: comp/api/api/apiimpl/internal/agent/agent_test.go (hunk 5 above)
Comment (jeremy-hanna) [reply comp/api/api/apiimpl/internal/agent/agent_test.go]: fixed in https://github.com/DataDog/datadog-agent/pull/23904/commits/22b5419f3bd559f234d94f099237a791045fc653

File: comp/api/api/apiimpl/api_test.go (hunk 4 above)
Comment (jeremy-hanna) [reply comp/api/api/apiimpl/api_test.go]: fixed in https://github.com/DataDog/datadog-agent/pull/23904/commits/6d1da76dde772349d03f441ee20f21b8e8eda144

**Review by spencergilbert — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Comment by pgimalac** [comment]
Adding `qa/no-code-change` as this PR only edits test or mock files.

**Pull request merged**